File: Atividades/palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool Verificador(const char *str) {
    int left = 0;
    int right = strlen(str) - 1;

    while (left < right) {
        if (str[left] != str[right]) {
            return false;
        }
        left++;
        right--;
    }
    return true;
}

int main() {
    char line[1000]; 

    while (1) {
        if (fgets(line, sizeof(line), stdin) == NULL) {
            break; 
        }

        line[strcspn(line, "\n")] = '\0';

        if (strcmp(line, "FIM") == 0) {
            break;
        }

        if (Verificador(line)) {
            printf("SIM\n");
        } else {
            printf("NAO\n");
        }
    }

    return 0;
}

File: Atividades/Palindromo.java
import java.util.*;

public class Palindromo {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (scanner.hasNextLine()) {
            String line = scanner.nextLine();
            if (line.equals("FIM")) {
                break;
            }
            System.out.println(Verificador(line) ? "SIM" : "NÃO");
        }
        scanner.close();
    }

    public static boolean Verificador(String str) {
        int left = 0;
        int right = str.length() - 1;
        
        while (left < right) {
            if (str.charAt(left) != str.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}
